govcd/catalog.go

```diff
@@ -663,9 +663,11 @@ func removeCatalogItemOnError(client *Client, vappTemplateLink *url.URL, itemNam
 			}
 		}
 
-		for _, task := range vAppTemplate.Tasks.Task {
-			if itemName == task.Owner.Name {
-				err = cancelTask(client, task.HREF)
+		for _, taskItem := range vAppTemplate.Tasks.Task {
+			if itemName == taskItem.Owner.Name {
+				task := NewTask(client)
+				task.Task = taskItem
+				err = task.CancelTask()
```

Comment:
Why are we creating a new task and then calling cancel on it as opposed to calling cancel on the taskItem straight? Same question for the change in media.go file.

Reply:
TaskItem is type represting xml, and Task is "object".